website/payments/admin.py

```diff
@@ -190,6 +245,53 @@ def process_wire_selected(self, request: HttpRequest, queryset: QuerySet) -> Non
 
     process_wire_selected.short_description = _("Process selected payments (wire)")
 
+    def add_to_new_batch(self, request: HttpRequest, queryset: QuerySet) -> None:
+        """Add selected TPAY payments to a new batch"""
+        tpays = queryset.filter(type=Payment.TPAY)
+        if len(tpays) > 0:
+            batch = Batch()
+            batch.save()
+            tpays.update(batch=batch)
```

Comment:
This whole part is doing business logic in admin.py, this should happen in services.py instead

Reply:
Should I then add them to services? They are quite short functions, so it seems weird.

Reply:
Yes they are short, but it would make it much easier to test these parts of the code.